src/psyclone/psyir/symbols/symboltable.py

```diff
@@ -180,40 +221,124 @@ def _normalize(key):
         new_key = key.lower()
         return new_key
 
-    def new_symbol_name(self, root_name=None, check_ancestors=True):
-        '''Create a symbol name that is not in this symbol table (if the
-        `check_ancestors` argument is False) or in this or any
-        ancestor symbol table (if the `check_ancestors` argument is
-        True). If the `root_name` argument is not supplied or if it is
+    def new_symbol(self, root_name=None, tag=None, shadowing=False,
+                   symbol_type=None, **symbol_init_args):
+        ''' Create a new symbol. Optional root_name and shadowing
+        arguments can be given to choose the name following the rules of
+        next_available_name(). An optional tag can also be given.
+        By default it creates a generic symbol but a symbol_type argument
+        and any additional initialization keyword arguments of this
+        sybmol_type can be provided to refine the created Symbol.
+
+        :param root_name: optional name to use when creating a new \
+            symbol name. This will be appended with an integer if the name \
+            clashes with an existing symbol name.
+        :type root_name: str or NoneType
+        :param str tag: optional tag identifier for the new symbol.
+        :param bool shadowing: optional logical flag indicating whether the \
+            name can be overlapping with a symbol in any of the ancestors \
+            symbol tables. Defaults to False.
+        :param symbol_type: class type of the new symbol.
+        :type symbol_type: Class type :py:class:`psyclone.psyir.symbols.Symbol`
```

Comment:
s/Class/class/ and maybe "subclass of :py:class:....."?

Reply:
Done substitution, but for the second suggestion:

Here what I mean is that it expects the Class object (which is of type: type) not the instance objects of this class or subclasses which we would normally write as "subclass of :py:class:..."
I modified to "type object of class (or subclasses) of :py:class:..."

Reply:
Oh yes, thanks.